### PR TITLE
btc-prov/wallet: btc-provider only publishes information to the speci…

### PR DESCRIPTION
**Pull request by tacryt-socryp**
…fic client that is asking for it now, except for new blocks and connection info.

Thanks for finding this issue @ixv!

### CODE REVIEW

**Comment by timlucmiptev** [comment]
Wow, that's brutal. The only thing I can say is that this is some of the very first logic written in the whole app, and my intuitions around subscriptions were a lot worse than. But still.....

Anyway, the fix to put ship name in subscription path looks good.

Is there a case to be made that this should be a poke back from the provider? I wouldn't let it hold up merging this, just been seeing this a lot lately.

**Comment by tacryt-socryp** [comment]
No, this is definitely better as a give than a poke, as it is logically a subscription.

File: pkg/arvo/app/btc-provider.hoon [hunk 1]
```diff
@@ -327,21 +334,24 @@
...
 ++  send-status
-  |=  =status
+  |=  [=status ship=(unit ship)]
   ^-  card
   %-  ?:  ?=(%new-block -.status)
         ~&(>> "%new-block: {<block.status>}" same)
       same
-  [%give %fact ~[/clients] %btc-provider-status !>(status)]
+  =-  [%give %fact ~[-] %btc-provider-status !>(status)]
+  ?~  ship  /clients
```
Comment (ixv) [inline pkg/arvo/app/btc-provider.hoon:343]: Do wires work that way? I've never seen this before.

Comment (tacryt-socryp) [reply pkg/arvo/app/btc-provider.hoon:343]: Yep! You can look at `/lib/push-hook` for more examples.

**Review by ixv — APPROVED** [review]

**Pull request merged**